File: src/model_v4/entities/SBS_EXTPROG.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryColumn } from "typeorm";
import { SBS_APKLASSE } from "./SBS_APKLASSE";


@Entity("SBS_EXTPROG",{schema:"sbsdb"})
@Index("FKBCD2838031F69485",["aPKLASSE_INDEX",])
export class SBS_EXTPROG {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"EXTPROG_INDEX"
        })
    EXTPROG_INDEX:string;
        

    @Column("varchar",{ 
        nullable:false,
        name:"EXTPROG"
        })
    EXTPROG:string;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"EXTPROG_NAME"
        })
    EXTPROG_NAME:string;
        

    @Column("varchar",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"EXTPROG_PAR"
        })
    EXTPROG_PAR:string | null;
        

    @Column("bigint",{ 
        nullable:true,
        default: () => "'NULL'",
        name:"FLAG"
        })
    FLAG:string | null;
        

   
    @ManyToOne(type=>SBS_APKLASSE, SBS_APKLASSE=>SBS_APKLASSE.sBS_EXTPROGs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'APKLASSE_INDEX'})
    aPKLASSE_INDEX:SBS_APKLASSE | null;

}

File: SbsdbServer/sbsdbv4/model/SbsExtprog.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_EXTPROG")]
    public partial class SbsExtprog
    {
        [Column("EXTPROG_INDEX", TypeName = "bigint(20)")]
        public long ExtprogIndex { get; set; }
        [Required]
        [Column("EXTPROG", TypeName = "varchar(255)")]
        public string Extprog { get; set; }
        [Required]
        [Column("EXTPROG_NAME", TypeName = "varchar(50)")]
        public string ExtprogName { get; set; }
        [Column("EXTPROG_PAR", TypeName = "varchar(255)")]
        public string ExtprogPar { get; set; }
        [Column("FLAG", TypeName = "bigint(20)")]
        public long? Flag { get; set; }
        [Column("APKLASSE_INDEX", TypeName = "bigint(20)")]
        public long? ApklasseIndex { get; set; }

        [ForeignKey("ApklasseIndex")]
        [InverseProperty("SbsExtprog")]
        public virtual SbsApklasse ApklasseIndexNavigation { get; set; }
    }
}
